doc/changelog.md

```diff
@@ -248,6 +248,10 @@
 
 <h3>Internal changes</h3>
 
+* llvm O2 and Enzyme passes are only run when needed (gradients presents). Async execution of QNodes triggers now triggers a
+   Coroutine lowering pass.
+  [(#968)](https://github.com/PennyLaneAI/catalyst/pull/968)
```

Comment:
It's a shame we don't have performance tracking / benchmarking available. It would allow us to say, compiler performance improved by "x% for non-diff programs" or something like that 😅